inc/amp.php

```diff
@@ -0,0 +1,139 @@
+	if ( ! is_amp() ) {
+		return $output;
+	}
+
+	if ( 'core/html' === $block['blockName'] ) {
+		$pym_src_regex = '~data-pym-src=[\'"]([^\'"]*)~';
+		$is_match      = preg_match( $pym_src_regex, $block['innerHTML'], $matches );
+		if ( ! $is_match ) {
+			return $output;
+		}
+
+		$pym_src = $matches[1];
+		return get_pym_ampiframe( $pym_src );
+	} elseif ( 'pym-shortcode/pym' === $block['blockName'] ) {
+		$pym_src = $block['attrs']['src'];
+		return get_pym_ampiframe( $pym_src, $block['attrs'] );		
+	}
+
+	return $output;
+}
+add_action( 'render_block', __NAMESPACE__ . '\convert_block_to_ampiframe', 10, 2 );
+
+/**
+ * Handle pym.js shortcode on AMP pages.
+ *
+ * @since  1.3.2.3
+ * @param  string $output HTML shortcode output.
+ * @param  string $tag Shortcode tag.
+ * @param  array  $attributes Shortcode attributes.
+ * @return string HTML shortcode output.
+ */
+function convert_shortcode_to_ampiframe( $output, $tag, $attributes ) {
+	if ( ! is_amp() || 'pym' !== $tag ) {
+		return $output;
+	}
+
+	if ( empty( $attributes['src'] ) ) {
+		return $output;
+	}
+
+	return get_pym_ampiframe( $attributes['src'], $attributes );
+}
+add_action( 'do_shortcode_tag', __NAMESPACE__ . '\convert_shortcode_to_ampiframe', 10, 3 );
+
+/**
+ * Build an amp-iframe out of pym.js iframe source. This is a pretty solid solution until native pym.js AMP compatibility.
+ *
+ * @since  1.3.2.3
+ * @see    https://github.com/ampproject/amphtml/issues/22714
+ * @param  string $src iframe src.
+ * @return string AMP-iframe HTML.
+ */
+function get_pym_ampiframe( $src, $atts = array() ) {
+	$src_domain_parts  = parse_url( $src );
+	$site_domain_parts = parse_url( get_site_url() );
+
+	if ( ! $src_domain_parts ) {
+		return '';
+	}
+
+
+	/**
+	 * Filter pym_shortcode_default_class allows setting the default class on embeds.
+	 *
+	 * @param String $default
+	 * @return String the default class name
+	 */
+	$default_class   = apply_filters( 'pym_shortcode_default_class', 'pym' );
+	$shortcode_class = empty( $atts['class'] ) ? '' : esc_attr( $atts['class'] );
+	$gutenberg_class = empty( $atts['className'] ) ? '' : esc_attr( $atts['className'] );
+	$align           = empty( $atts['align'] ) ? '' : 'align' . esc_attr( $atts['align'] );
+	$actual_classes  = implode( ' ', array(
+		$default_class,
+		$shortcode_class,
+		$gutenberg_class,
+		$align,
+	) );
+	$id              = empty( $atts['id'] ) ? '' : esc_attr( $atts['id'] );
+
+	$sandbox = 'allow-scripts';
+	if ( strcasecmp( $src_domain_parts['host'], $site_domain_parts['host'] ) ) {
+		$sandbox .= ' allow-same-origin';
+	}
+
+	ob_start();
+	?>
+	<div class='<?php echo esc_attr( $actual_classes ); ?>' id='<?php echo esc_attr( $id ); ?>'>
+		<amp-iframe 
+			src='<?php echo esc_url( $src ); ?>'
+			layout='intrinsic'
+			width='1200'
+			height='1200'
+			sandbox='<?php echo esc_attr( $sandbox ); ?>'
+			frameborder='0'
+			resizable
+		>
+			<div placeholder><?php esc_html_e( 'Interactive graphic', 'pym-embeds' ); ?></div>
```

Comment:
Thanks for helping us catch https://github.com/INN/pym-shortcode/issues/67; it's not something that should be addressed in this PR but it is good to know about.

Reply:
Should I remove the text domains used in this PR and just let it fall back to the default textdomain? Or will you handle that later as part of the general text domain cleanup mentioned in #67?